build.gradle

```diff
@@ -22,85 +21,131 @@ group 'io.xh'
+    developmentOnly
+    runtimeClasspath {
+        extendsFrom developmentOnly
+    }
 }
 
 boolean inlineHoist = runHoistInline == 'true'
 if (inlineHoist) {
     println "${appName}: running with Hoist Core INLINE...."
+    grails.plugins {
+        implementation project(":hoist-core")
+    }
 } else {
     println "${appName}: running with Hoist Core PACKAGED at v${hoistCoreVersion}...."
+    dependencies {
+        implementation "io.xh:hoist-core:$hoistCoreVersion"
+    }
 }
 
 dependencies {
-    compile "org.springframework.boot:spring-boot-starter-logging"
-    compile "org.springframework.boot:spring-boot-autoconfigure"
-    compile "org.grails:grails-core"
-    compile "org.springframework.boot:spring-boot-starter-actuator"
-    provided "org.springframework.boot:spring-boot-starter-tomcat"
-
-    compile "org.grails:grails-web-boot"
-    compile "org.grails:grails-logging"
-    compile "org.grails:grails-plugin-rest"
-    compile "org.grails:grails-plugin-databinding"
-    compile "org.grails:grails-plugin-i18n"
-    compile "org.grails:grails-plugin-services"
-    compile "org.grails:grails-plugin-url-mappings"
-    compile "org.grails:grails-plugin-interceptors"
-
-    compile "org.grails.plugins:gsp"
-    compile "org.grails.plugins:async"
-    compile "org.grails.plugins:events"
-
-    compile "org.grails.plugins:hibernate5"
-    compile "org.hibernate:hibernate-core:5.1.5.Final"
-    compile "org.hibernate:hibernate-ehcache:5.1.5.Final"
-
-    console "org.grails:grails-console"
-    profile "org.grails.profiles:web"
-
-    compile "org.springframework:spring-websocket"
-
-    // For JWT token validation.
-    compile "org.bitbucket.b_c:jose4j:0.7.2"
-
-    runtime "org.apache.tomcat:tomcat-jdbc"
-    runtime "mysql:mysql-connector-java:5.1.47"
-
-    if (!inlineHoist) {
-        // Include Hoist as versioned external dependency when building a WAR for release
-        compile "io.xh:hoist-core:$hoistCoreVersion"
-    }
+    implementation "org.bitbucket.b_c:jose4j:0.7.2"
+    runtimeOnly "mysql:mysql-connector-java:5.1.47"
+
+    // Hot-reload support
+    developmentOnly(
+        "org.springframework.boot:spring-boot-devtools",
+        "io.methvin:directory-watcher:0.15.0"
+    )
 }
 
 // Avoid unexpected errors with overly-long classpath on Windows development machines.
 grails {
     pathingJar = true
 }
 
-if (inlineHoist) {
-    // Include Hoist as an inline plugin in multi-project mode when running in development
-    grails {
-        plugins {
-            compile project(":hoist-core")
+
+// Tasks to create distribution bundles that include built client-assets, as well as (for the
+// first task) the WAR. This creates a self-contained archive with both sides of the build output.
+// NOTE - these rely on the client-side build having been completed (outside of Gradle) by Yarn and
+// Webpack. A future project could be to orchestrate the Webpack build from within Gradle as well.
+apply plugin:'distribution'
```

Comment:
I would be happy to remove all of this complexity around publishing from the Gradle file. We added this as part of research for a client who wanted some custom artifact publishing for app builds, but I am not sure they are actually using it, and either way I don't think it's standard enough for us to bother including in Toolbox as an example.

Happy to let the Docker container images be our versioned outputs from the builds.